tests/saw/verify_HMAC_bad_magic_mod.saw

```diff
@@ -0,0 +1,18 @@
+
+include "HMAC/spec/verify_HMAC.saw";
+
+/*
+ * The sha_bad_magic_mod patch changes the highest_32_bit value in s2n_hmac_update from 4294949760 to 4294949761. To
+ * ensure SAW tests this function with the correct values when this patch is applied, the highest_32_bit value is
+ * increased in SAW as well, via this config.
+ */
+let sha256_bad_magic_mod_cfg =
+  { name            = "SHA256_bad_magic_mod"
+  , hmac_alg        = {{ S2N_HMAC_SHA256 }}
+  , digest_size     = 32
+  , block_size      = 64
+  , hash_block_size = 64
+  , highest_32_bit  = 4294949761
```

Comment:
Even by patching this value, do you still get the expected failure in the test case?

Reply:
I believe so, yes. When running this test with the value changed in SAW and in C, we get the following error:
```
Subgoal failed: s2n_hmac_update safety assertion:
s2n-tls/tests/saw/verify_HMAC_bad_magic_mod.saw:18:1: error: in _SAW_verify_prestate
Literal equality postcondition
```
which seems correct.

The full log is here: https://us-west-2.console.aws.amazon.com/codesuite/codebuild/024603541914/projects/s2nGeneralBatch/build/s2nGeneralBatch%3Aea3d0c29-6e37-420d-9299-50e2fad6d7fa?region=us-west-2#

Reply:
Do you know the reason why the proof fails after patching this value? I ask because this will clarify what this failure test is really testing for.

Reply:
Slight clarification, the Cryptol spec isn't particularly optimized, instead the C implementation is "de-optimized" in order to give constant time behavior. The original purpose of this test was to ensure that if a bad "magic mod" (aka highest_32_bit) number were selected that would fail the proof.

Once the CBMC conditions were added, the failure mode changed to triggering the overflow condition. Without it, the overflow still happens, but exhibits an incorrect output rather than as an assertion failure.